### PR TITLE
exe-config: fix ghcjs config gathering

### PR DESCRIPTION
**Pull request by tomsmalley**
Previously the keys in the map would evaluate to `<undefined>`, leading to really weird behaviour: `getConfig` returning the same config no matter the path, and, it seems, hangs under some circumstances

### CODE REVIEW

**Comment by 3noch** [comment]
If I could approve this twice I would.

**Comment by xplat** [comment]
if i could approve it at all i would ...

**Comment by 3noch** [comment]
If I could approve you to give approval I would...

**Comment by 3noch** [comment]
Turns out I can. :joy:

File: lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs [hunk 1]
```diff
@@ -30,7 +30,8 @@ getConfigs = do
       Just htmlE -> return htmlE
     dataset <- getDataset e
     (,)
-      <$> get dataset ("obelisk-executable-config-inject-key" :: Text)
+      -- the key is camelCased: https://html.spec.whatwg.org/multipage/dom.html#dom-dataset
```
Comment (ali-abrar) [inline lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs]: delightful

**Pull request merged**